internal/backends/backend_test.go

```diff
@@ -64,3 +64,71 @@ func TestVersion(t *testing.T) {
 		})
 	}
 }
+
+func TestListDatabases(t *testing.T) {
+	t.Parallel()
+
+	ctx := conninfo.Ctx(testutil.Ctx(t), conninfo.New())
+
+	for name, b := range testBackends(t) {
+		name, b := name, b
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+			// setup 3 DB with 1 collections each. random order of db name also ensure sorting test.
+			dbNames := []string{"testDB2", "testDB1", "testDB3"}
+			collectionName := "testCollection"
+
+			testDB, err := b.Database(dbNames[0])
+			require.NoError(t, err)
+			err = testDB.CreateCollection(ctx, &backends.CreateCollectionParams{Name: collectionName})
+			require.NoError(t, err)
+			defer b.DropDatabase(ctx, &backends.DropDatabaseParams{Name: dbNames[0]}) //nolint:errcheck // safe to ignore
+
+			testDB, err = b.Database(dbNames[1])
+			require.NoError(t, err)
+			err = testDB.CreateCollection(ctx, &backends.CreateCollectionParams{Name: collectionName})
+			require.NoError(t, err)
+			defer b.DropDatabase(ctx, &backends.DropDatabaseParams{Name: dbNames[1]}) //nolint:errcheck // safe to ignore
+
+			testDB, err = b.Database(dbNames[2])
+			require.NoError(t, err)
+			err = testDB.CreateCollection(ctx, &backends.CreateCollectionParams{Name: collectionName})
+			require.NoError(t, err)
+			defer b.DropDatabase(ctx, &backends.DropDatabaseParams{Name: dbNames[2]}) //nolint:errcheck // safe to ignore
+
+			t.Run("TestingListDatabases", func(t *testing.T) {
+				t.Run("ListDatabases with specific name", func(t *testing.T) {
+					dbRes, err := b.ListDatabases(ctx, &backends.ListDatabasesParams{Name: dbNames[2]})
+					require.NoError(t, err)
+					require.Equal(t, 1, len(dbRes.Databases), "expected len 1 , since only 1 db with name testDB3")
+					require.Equal(t, dbNames[2], dbRes.Databases[0].Name, "expected name testDB3")
+				})
+
+				t.Run("ListDatabases with dummy name", func(t *testing.T) {
+					dbRes, err := b.ListDatabases(ctx, &backends.ListDatabasesParams{Name: "dummy"})
```

Comment:
Those subtests could run in parallel, so let's add `t.Parallel`

Reply:
Ok. done.